src/components/rating/rating.tsx

```diff
@@ -222,11 +221,48 @@ export class Rating implements LabelableComponent, FormComponent, InteractiveCom
     this.setFocus();
   }
 
+  /**
+   * update `value` property of the rating component
+   *
+   * @param value value used to update `value` property
+   * @internal
+   */
   private updateValue(value: number) {
     this.value = value;
     this.calciteRatingChange.emit({ value });
   }
 
+  /**
+   * register keyboard input and reset `value` property to `0` if user enters 'Space' or 'Enter'
+   *
+   * @param event keyboard event
+   * @internal
+   */
+  private onKeyboardPressed(event: KeyboardEvent): void {
+    // click is fired from the the component's input, so we treat this as an internal event
+    event.stopPropagation();
+    event.preventDefault();
+    if (event.key === "Enter" || event.code === "Space") {
+      this.updateValue(0);
```

Comment:
Can you add a new `required` prop that prevents clearing the selected rating?

Reply:
> Can you add a new `required` prop that prevents clearing the selected rating?

@benelan I did not understand. Could you please give me more context on this `required` props?

Reply:
We want to add a new property named `required`. When the prop is `true`, the value of the rating should not be clearable (set to zero) using any of the methods you added in this PR (keydown and click). So for the code above, it would be something like: 
`if (!this.required && ( event.key === " " || event.key=== "Enter")) { ...`

Reply:
> We want to add a new property named `required`. When the prop is `true`, the value of the rating should not be clearable (set to zero) using any of the methods you added in this PR (keydown and click). So for the code above, it would be something like: `if (!this.required && ( event.key === " " || event.key=== "Enter")) { ...`

@benelan we already have a `required` prop in the rating component.
https://github.com/Esri/calcite-components/blob/ec144ebc01dfb4803b1d2336c53a496438a2a760/src/components/rating/rating.tsx#L84

Should I create a new prop with different variable name or should I use this already defined `required` prop for the checks you suggested me earlier?

Reply:
Sorry, not a new property. Use the existing `required` prop for the checks. Thanks!